### ISSUE TITLE
Issue with OpenStreetMap tiles over https

### ISSUE THREAD

**Issue by astrofrog** [WorldWideTelescope/wwt-squid-proxy#1]
The following does not work:

https://beta.worldwidetelescope.org/webserviceproxy.aspx?targeturl=https%3A%2F%2Fa.tile.openstreetmap.org%2F6%2F44%2F22.png

but using http does:

https://beta.worldwidetelescope.org/webserviceproxy.aspx?targeturl=http%3A%2F%2Fa.tile.openstreetmap.org%2F6%2F44%2F22.png

This doesn't seem to be a generic https issue since other https URLs work:

https://beta.worldwidetelescope.org/webserviceproxy.aspx?targeturl=https%3A%2F%2Favatars3.githubusercontent.com/u/314716?s=460&u=2ddf7ef795245a09841c4c3cec44d5f2e1dc4265&v=4

I tried debugging the squid proxy locally but couldn't find the origin of the **Read error**.

**Comment by astrofrog** [comment]
As an experiment, I added:

```
    $targeturl =~ s/https:\/\/a.tile/http:\/\/a.tile/g;
```

after the ``uri_unescape`` line in ``rewriter.pl`` and it then works fine. If we want to keep this kind of hack out of the proxy (which I assume we do) we could modify the web client to never request OSM tiles over https?

**Comment by pkgw** [comment]
Yeah, I think the correct approach here would be to add a "HTTP-only" category to DomainHandling in URLHelpers.cs in the engine, forcing filtering through the proxy if the URL is requested on an HTTP backend.

**Comment by pkgw** [comment]
Hm, this is not quite what I expected. First of all, it looks like we can load the OpenStreetMap tiles in the webclient using either HTTP or HTTPS — the proxy isn't necessary.

That being said, I do consistently get errors when trying to load tiles through the proxy. But, OSM does support HTTPS under the hood, so it's not as simple as forcing HTTP. Squid reports a "read error" and says that this is the request it issued:

```
CacheHost: 2d54ec159464
ErrPage: ERR_READ_ERROR
Err: [none]
TimeStamp: Mon, 03 Aug 2020 22:18:35 GMT

ClientIP: 172.16.0.1
ServerIP: a.tile.openstreetmap.org

HTTP Request:
GET /6/44/22.png HTTP/1.1
X-Client-IP: 13.87.152.132
X-Client-Port: 4355
Connection: Keep-Alive
Accept: */*
Max-Forwards: 10
User-Agent: curl/7.71.1
X-ORIGINAL-HOST: worldwidetelescope.org
X-FORWARDED-PROTO: https
X-FORWARDED-PORT: 443
X-Forwarded-For: 76.119.232.243:41148, 13.87.152.132:4355
X-Original-URL: /webserviceproxy.aspx?targeturl=https%3A%2F%2Fa.tile.openstreetmap.org%2F6%2F44%2F22.png
X-WAWS-Unencoded-URL: /webserviceproxy.aspx?targeturl=https%3A%2F%2Fa.tile.openstreetmap.org%2F6%2F44%2F22.png
CLIENT-IP: 13.87.152.132:4355
X-ARR-LOG-ID: 5b6b4ace-5693-4602-9b90-45224d1e7aef
DISGUISED-HOST: wwtproxy.azurewebsites.net:80
X-SITE-DEPLOYMENT-ID: wwtproxy
WAS-DEFAULT-HOSTNAME: wwtproxy.azurewebsites.net
Host: wwtproxy.azurewebsites.net
```

I tried curl-ing with a few custom headers but it's not obvious what exactly is triggering the issue here.
